### PR TITLE
Make NotificationMessage an Interface

### PR DESCRIPTION
**Pull request by marstr**
Hiding all the rich details from Firebase's `RemoteMessage` and `RemoteMessage.Notification` is whack, but we have other sources like `Intent`s that we get from the system tray. Instead of throwing away all the extra data, we can work behind an interface.

There are a few ways we could handle RemoteMessage and Intent specific implementations. We're still working out some of those details. More to come in a future PR.

### CODE REVIEW

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/INotification.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import java.util.Map;
+
+/**
+ * Wraps the notifications that may be delivered via any channel.
+ */
+public interface INotification {
```
Comment (brannon) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/INotification.java:8]: Java doesn't typically use the "I" prefix on interfaces, right?

Should we keep this named `NotificationMessage` and change the name of the current class implementing it?

Comment (mpodwysocki) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/INotification.java:8]: Agreed, we should have `NotificationMessage` as the interface and something like `FirebaseNotificationMessage` or whatever we want to call it as the implementation

Comment (marstr) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/INotification.java]: Fixed in de6ef36cd9fcc336e182ad3b7e1fd64432329c90

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/BasicNotificationMessage.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * Instantiates a basic notification.
+     * @param title The primary text associated with this notification.
+     * @param message The secondary text associated with this notification.
+     * @param data The key/value pairs of data associated with this notification.
+     */
+    public BasicNotificationMessage(String title, String message, @NonNull Map<String, String> data) {
+        mTitle = title;
+        mMessage = message;
+        mCustomData = data;
```
Comment (mpodwysocki) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/BasicNotificationMessage.java:36]: Should this be called custom data still?

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/BasicNotificationMessage.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Notification message. Will be null when Notification is received from click in system tray.
+     */
+    private final String mMessage;
+
+    /**
+     * Key/value pairs sent with a Push Notification.
+     */
+    private final Map<String, String> mCustomData;
```
Comment (brannon) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/BasicNotificationMessage.java:25]: Just to be consistent, we should name this `mData`.

Comment (marstr) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/BasicNotificationMessage.java]: Fixed in c0beaae34fce3e8c21d6a61ee80f0ee43147433e

**Comment by marstr** [comment]
Not sure why GH says I dismissed the approvals. But I went ahead with these changes.

**Pull request merged**